app/screens/onboarding/slide.tsx

```diff
@@ -0,0 +1,221 @@
+        textAlign: 'center',
+        paddingHorizontal: 20,
+        maxWidth: ONBOARDING_CONTENT_MAX_WIDTH,
+        marginBottom: 12,
+    },
+    fontTitle: {
+        marginTop: 32,
+        ...typography('Heading', 1000, 'SemiBold'),
+    },
+    fontFirstTitle: {
+        ...typography('Heading', 1200, 'SemiBold'),
+        paddingTop: 48,
+        letterSpacing: -1,
+    },
+    widthLastSlide: {
+        paddingHorizontal: 50,
+    },
+    firstSlideInitialPosition: {
+        left: 200,
+        opacity: 0,
+    },
+    description: {
+        textAlign: 'center',
+        paddingHorizontal: 20,
+        color: changeOpacity(theme.centerChannelColor, 0.64),
+        ...typography('Body', 200, 'Regular'),
+        maxWidth: ONBOARDING_CONTENT_MAX_WIDTH,
+    },
+    itemContainer: {
+        flex: 1,
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+}));
+
+const FIRST_SLIDE = 0;
+
+const SlideItem = ({theme, item, scrollX, index, lastSlideIndex}: Props) => {
+    const {width} = useWindowDimensions();
+    const styles = getStyleSheet(theme);
+
+    /**
+     * Code used to animate the first image load
+     */
+    const [firstLoad, setFirstLoad] = useState(true);
+
+    const initialImagePosition = useSharedValue(width);
+    const initialTitlePosition = useSharedValue(width);
+    const initialDescriptionPosition = useSharedValue(width);
+
+    const initialElementsOpacity = useSharedValue(0);
+
+    useEffect(() => {
+        if (index === FIRST_SLIDE) {
+            initialImagePosition.value = withTiming(0, {duration: 600});
+            initialTitlePosition.value = withTiming(0, {duration: 800});
+            initialDescriptionPosition.value = withTiming(0, {duration: 1000});
+
+            initialElementsOpacity.value = withTiming(1, {duration: 1000});
+            setFirstLoad(false);
+        }
+    }, []);
+
+    const translateFirstImageOnLoad = useAnimatedStyle(() => {
+        return {
+            transform: [{
+                translateX: initialImagePosition.value,
+            }],
+            opacity: initialElementsOpacity.value,
+        };
+    });
+
+    const translateFirstTitleOnLoad = useAnimatedStyle(() => {
+        return {
+            transform: [{
+                translateX: initialTitlePosition.value,
+            }],
+            opacity: initialElementsOpacity.value,
+        };
+    });
+
+    const translateFirstDescriptionOnLoad = useAnimatedStyle(() => {
+        return {
+            transform: [{
+                translateX: initialDescriptionPosition.value,
+            }],
+            opacity: initialElementsOpacity.value,
+        };
+    });// end of code for animating first image load
+
+    const inputRange = [(index - 1) * width, index * width, (index + 1) * width];
+
+    const translateImage = useAnimatedStyle(() => {
+        const translateImageInterpolate = interpolate(
+            scrollX.value,
+            inputRange,
```

Comment:
Not sure what the implications are, but `useAnimatedStyle` recreates the animated style if any of the closures change. `inputRange` is changing on every render.

Not sure if this is a problem, so it wold be interesting to consider.

Reply:
This could be causing a few frame drops, not a massive problem but maybe can be improved by having the inputRange as a shared value?

Reply:
Having it memoized, or passing a list of dependencies should be enough.